es/ReactTelephoneInput.js

```diff
@@ -173,7 +173,13 @@ export var ReactTelephoneInput = createReactClass({
         return !isEqual(nextProps, this.props) || !isEqual(nextState, this.state);
     },
     componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
-        this.setState(this._mapPropsToState(nextProps));
+        var prevFormattedNumber = this.state.formattedNumber;
+        this.setState(this._mapPropsToState(nextProps), function () {
+            // If formatted number has changed, call on change
+            if (this.state.formattedNumber !== prevFormattedNumber) {
+                this.props.onChange(this.state.formattedNumber, this.state.selectedCountry);
+            }
```

Comment:
As this component maintains its own state, if there are changes to the value of the telephone field that were caused by the incoming props these changes will be applied in `_mapPropsToState`. Prior to this fix, such a change in the state will not call `onChange`. This fixes it.

Reply:
@mukeshsoni any objections to this change? Perhaps this was suggested in the past?

Reply:
componentWillReceiveProps is doing a lot in this component. I shouldn't have done this in the first place. This piece of code has been the source of a lot of bugs in this component. I am thinking about completely removing componentWillReceiveProps and give an alternate solution making it a controlled/uncontrolled/(controlled + uncontrolled) component